### DISCUSSION TITLE
Migrating from sentry/sentry-laravel 3.8.2 -> 4.2.0

### DISCUSSION THREAD

**Discussion by gwidgery** [getsentry/sentry-laravel#855] [Q&A] [answered]
Last week I updated sentry to the latest version but stopped receiving errors into sentry. I have just reverted back to the previous working version. Any suggestions on what I'm doing wrong?

I was trying to follow the installation guide here:
[https://docs.sentry.io/platforms/php/guides/laravel/](https://docs.sentry.io/platforms/php/guides/laravel/)

Handler.php

Working Code:

```
public function report(Throwable $exception)
    {
        if (app()->bound('sentry') && $this->shouldReport($exception)) {
            app('sentry')->captureException($exception);
        }
        parent::report($exception);
    }
```

Changed to this for latest version, but then no longer receiving errors:

```
public function register(): void
{
    $this->reportable(function (Throwable $e) {
            Integration::captureUnhandledException($e);
        });
        parent::register(); // TODO: Change the autogenerated stub
}
```

**Comment by cleptric** [comment]
Which Laravel version are you using?

**Reply by gwidgery** [reply]
Laravel 10

**Reply by cleptric** [reply]
Did you maybe forget to upgrade your laravel template along the line?
Looking at Laravel 10, there should be a register method https://github.com/laravel/laravel/blob/10.x/app/Exceptions/Handler.php

**Comment by stayallive** [comment]
After the migration, did you also run `php artisan sentry:test` to see if anything might be not working correctly?

**Reply by gwidgery** [reply]
Yes, I ran `php artisan sentry:test` and that filed the test exception. It only seems to be the code generated exceptions that weren't working.

I ran this code
```
Route::get('/debug-sentry', function () {
    throw new Exception('My first Sentry error!');
});
```

and the exception renders in the browser, but nothing was sent to sentry.

**Reply by stayallive** [reply, marked as answer]
Can you try adding this to your `sentry.php` next to your other options:

```php
	'logger' => new \Sentry\Logger\DebugFileLogger(storage_path('logs/sentry.log')),
```

**Note:** if you have config caching, disable it for a bit since this won't work with that!

This should write to `storage/logs/sentry.log` any internal logs from the SDK that might tell you what is not working. My first guess is that you don't have the `cURL` extension enabled for your web requests but do have it enabled on the CLI. The logs _should_ surface that if that is the case.

**Reply by gwidgery** [reply]
Thanks - that's fixed it. Enabled logging and saw that the `cURL` extension wasn't enabled.
